src/IO/AssimpLoader/AssimpCameraDataLoader.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef RADIUMENGINE_ASSIMP_CAMERA_DATA_LOADER_HPP
+#define RADIUMENGINE_ASSIMP_CAMERA_DATA_LOADER_HPP
+
+#include <Core/File/DataLoader.hpp>
+#include <Core/Math/LinearAlgebra.hpp>
+#include <IO/RaIO.hpp>
+
+#include <set>
+
+struct aiScene;
+struct aiCamera;
+struct aiNode;
+
+namespace Ra {
+namespace Engine {
+class Camera;
+} // namespace Engine
+
+namespace Asset {
+class CameraData;
+} // namespace Asset
+} // namespace Ra
+
+namespace Ra {
+namespace IO {
+
+/// The AssimpCameraDataLoader converts camera data from the Assimp format
+/// to the Asset::CameraData format.
+class RA_IO_API AssimpCameraDataLoader : public Asset::DataLoader<Asset::CameraData> {
+  public:
+    AssimpCameraDataLoader( const std::string& filepath, const bool VERBOSE_MODE = false );
+
+    ~AssimpCameraDataLoader();
+
+    /// Convert all the camera data from \p scene into \p data.
+    void loadData( const aiScene* scene,
+                   std::vector<std::unique_ptr<Asset::CameraData>>& data ) override;
+
+  protected:
+    /// Return true if the given scene has camera data.
+    inline bool sceneHasCamera( const aiScene* scene ) const;
+
+    /// Return the number of cameras in the given scene.
+    uint sceneCameraSize( const aiScene* scene ) const;
+
+    /// Fill \p data with the CameraData from \p camera.
+    void loadCameraData( const aiScene* scene, const aiCamera& camera, Asset::CameraData& data );
+
+    /// Fill \p data with the camera name from \p camera.
+    void fetchName( const aiCamera& camera, Asset::CameraData& data ) const;
+
+    /// Return the light transformation, in world space, for \p data from \p the scene.
+    Core::Matrix4 loadCameraFrame( const aiScene* scene, const Core::Matrix4& parentFrame,
+                                   Asset::CameraData& data ) const;
```

Comment:
as the parentFrame might be shared between several objects, this ensure that the method will not change the frame. I prefer to let this const rather than expect that the caller will give a copy of an eventually shared parentFrame.

Reply:
Of course parent frame must be kept as const., this not what I suggest ;)

Reply:
Well, if you directly build it and return a local object, or, under some circumstances, if you build a local object, modify it then return the modified object, the C++ compiler must activate, copy elision mechanism. The local variable is directly associated with the object (in the memory) that will receive the result of the return (if it exist, otherwise, it is a temporary) so that thereis no copy nor move.
see https://en.cppreference.com/w/cpp/language/copy_elision for a complete doculentation about copy elision.